### PR TITLE
Move runjs into tests directory

### PR DESCRIPTION
**Pull request by sbc100**
Most of the code in this file relates to running of tests and its not
part of the toolchain itself.

### CODE REVIEW

File: tests/fuzz/csmith_driver.py [hunk 1]
```diff
@@ -176,8 +176,8 @@ def try_js(args=[]):
   def execute_js(engine):
     print('(run in %s)' % engine)
     try:
-      js = jsrun.run_js(filename + '.js', engine=shutil.NODE_JS, check_timeout=True, assert_returncode=None)
-    except CalledProcessError:
+      js = shared.timeout_run(Popen(shared.NODE_JS + [filename + '.js'], stdout=PIPE, stderr=PIPE), 15 * 60)
```
Comment (sbc100) [inline tests/fuzz/csmith_driver.py:179]: this is the only line I'm not totally sure about..

Comment (kripken) [reply tests/fuzz/csmith_driver.py:179]: The timeout here is probably larger than it needs to be, could be 5 seconds or so probaly.

Also worth running this to see if it works. However, I'm not sure any of us has run the csmith driver for a while, so maybe it broke meanwhile.

File: tools/shared.py [hunk 2]
```diff
@@ -256,6 +257,25 @@ def is_exe(fpath):
   return None
 
 
+# Only used by tests and by ctor_evaller.py.   Once fastcomp is remove
```
Comment (kripken) [inline tools/shared.py:260]:
```suggestion
# Only used by tests and by ctor_evaller.py.   Once fastcomp is removed
```

File: tools/shared.py [hunk 3]
```diff
@@ -256,6 +257,25 @@ def is_exe(fpath):
...
+# this can most likely be moved in the tests/jsrun.py.
```
Comment (kripken) [inline tools/shared.py:261]:
```suggestion
# this can most likely be moved into the tests/jsrun.py.
```

File: tests/fuzz/csmith_driver.py (hunk 1 above)
Comment (sbc100) [reply tests/fuzz/csmith_driver.py:179]: Does it need timeout?  I guess its there in case the generated code doesn't terminate?

**Comment by sbc100** [comment]
> My one concern is the rename of `hello_world.js` which seems good but has some risk that on CI or elsewhere we depend on it for a sanity check. In fact maybe we have it in src/ because we don't want to assume the test dir is present in all places we need such a check?

But there is only one place we run the check and I made it inline using just:  -e  'console.log("hello")'.

Outside of the test harness what purpose do you see hello_world.js serving ?   Also, BTW we do still ship the tests directory to all users.. we just trim the third_party subdirectory.

File: tests/fuzz/csmith_driver.py (hunk 1 above)
Comment (kripken) [reply tests/fuzz/csmith_driver.py:179]: Yes, csmith code may not terminate (unlike the binaryen fuzzer output, which guarantees it)

**Pull request merged**